lib/HttpSignatureStrategy.js

```diff
@@ -211,13 +216,46 @@ function getDidPublicKey(publicKey, callback) {
   }, callback);
 }
 
+// 1. resolve key ID => should contain public key info including key material
+// (and owner)
+// 2. resolve owner ID and get identity info and list of public keys
+// 3. make sure key ID is listed in list of keys -- if so, verified
 function getHttpsPublicKey(publicKey, callback) {
-  // TODO: implement
-  callback(new BedrockError(
-    'URL scheme "https" is not supported.',
-    'HttpSignature.UnsupportedUrlScheme', {
-      'public': true,
-      httpStatusCode: 400,
-      scheme: 'https'
-    }));
+  async.auto({
+    dereferencePublicKey: function(callback) {
+      request.get(publicKey, function(err, res, body) {
+        callback(err, res.body);
+      });
+    },
+    dereferenceOwner: ['dereferencePublicKey', function(callback, results) {
+      var dereferencedKey = results.dereferencePublicKey;
+      var owner = dereferencedKey.owner;
+      request.get(owner, function(err, res, body) {
+        callback(err, res.body);
+      });
+    }],
+    verify: ['dereferenceOwner', function(callback, results) {
+      var dereferencedOwner = results.dereferenceOwner;
+      var dereferencedKey = results.dereferencePublicKey;
+      var ownerKeys = jsonld.getValues(dereferencedOwner, 'publicKey');
+      var found = false;
+      ownerKeys.forEach(function(key) {
+        if(key.id === dereferencedKey.id) {
+          found = true;
+          return false;
```

Comment:
Just a note that `return false;` doesn't actually break the `forEach` loop or anything. You'll need to use (misuse?) `.some` to break a loop like this -- or just use the traditional `for(var i = 0; i < ownerKeys.length; ++i) { ... break`